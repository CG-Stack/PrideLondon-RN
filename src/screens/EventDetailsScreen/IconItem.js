// @flow
import React from "react";
import { View, StyleSheet } from "react-native";
import type { Node } from "react";

type IconItemProps = {|
  icon: Node,
  children: Node
|};

const IconItem = ({ icon, children }: IconItemProps) => (
  <View style={styles.iconItem}>
    <View style={styles.icon}>{icon}</View>
    <View style={styles.item}>{children}</View>
  </View>
);

const styles = StyleSheet.create({
  iconItem: {
    flex: 1,
    flexDirection: "row"
  },
  icon: {
    width: 40,
    height: 32,
    marginRight: 12
  },
  item: {
    flex: 1,
    paddingTop: 2
  }
});

export default IconItem;
