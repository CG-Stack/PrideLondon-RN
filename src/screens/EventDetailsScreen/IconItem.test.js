// @flow
import React from "react";
import { Text } from "react-native";
import { shallow } from "enzyme";
import IconItem from "./IconItem";

it("renders correctly", () => {
  const output = shallow(
    <IconItem icon={<Text>Some icon</Text>}>
      <Text type="h4">some thing</Text>
    </IconItem>
  );
  expect(output).toMatchSnapshot();
});
