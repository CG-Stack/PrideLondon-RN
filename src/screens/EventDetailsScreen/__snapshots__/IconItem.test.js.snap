// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "height": 20,
        "marginRight": 18,
        "width": 40,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Some icon
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      allowFontScaling={true}
      markdown={false}
      style={
        Array [
          Object {
            "color": "#2d2f7f",
          },
          Object {},
        ]
      }
      type="h4"
    >
      some thing
    </Text>
  </View>
</View>
`;
